src/server/ServerHandler.java

```diff
@@ -64,6 +80,110 @@ public ServerHandler(Socket socket, String username) {
         }
     }
 
+    /**
+     * Main request handler that processes client requests and returns appropriate responses.
+     *
+     * @param request the request object received from the client
+     * @return a response object based on the request type
+     */
+    public Response handleRequest(Request request) {
+        if (request == null) {
```

Comment:
Let's add some logging if the request is null

```suggestion
        if (request == null) {
            LOGGER.warn("Received null request");
```